telemetry/src/main/java/com/newrelic/telemetry/TelemetryClient.java

```diff
@@ -183,24 +208,65 @@ public void shutdown() {
    */
   public static TelemetryClient create(
       Supplier<HttpPoster> httpPosterCreator, String insertApiKey) {
-    MetricBatchSender metricBatchSender =
-        MetricBatchSender.create(
-            MetricBatchSenderFactory.fromHttpImplementation(httpPosterCreator::get)
-                .configureWith(insertApiKey)
-                .build());
-
-    SpanBatchSender spanBatchSender =
-        SpanBatchSender.create(
-            SpanBatchSenderFactory.fromHttpImplementation(httpPosterCreator::get)
-                .configureWith(insertApiKey)
-                .build());
-
-    EventBatchSender eventBatchSender =
-        EventBatchSender.create(
-            EventBatchSenderFactory.fromHttpImplementation(httpPosterCreator::get)
-                .configureWith(insertApiKey)
-                .build());
-
-    return new TelemetryClient(metricBatchSender, spanBatchSender, eventBatchSender);
+    return Builder.defaultBuilder(httpPosterCreator, insertApiKey).build();
+  }
+
+  public static class Builder {
+    private MetricBatchSender metricBatchSender;
+    private SpanBatchSender spanBatchSender;
+    private EventBatchSender eventBatchSender;
+    private long shutdownSeconds;
+
+    public Builder metricBatchSender(MetricBatchSender metricBatchSender) {
+      this.metricBatchSender = metricBatchSender;
+      return this;
+    }
+
+    public Builder spanBatchSender(SpanBatchSender spanBatchSender) {
+      this.spanBatchSender = spanBatchSender;
+      return this;
+    }
+
+    public Builder eventBatchSender(EventBatchSender eventBatchSender) {
+      this.eventBatchSender = eventBatchSender;
+      return this;
+    }
+
+    public Builder shutdownSeconds(long shutdownSeconds) {
+      this.shutdownSeconds = shutdownSeconds;
+      return this;
+    }
+
+    public TelemetryClient build() {
+      return new TelemetryClient(this);
+    }
+
+    public static Builder defaultBuilder(
+        Supplier<HttpPoster> httpPosterCreator, String insertApiKey) {
+      MetricBatchSender metricBatchSender =
+          MetricBatchSender.create(
+              MetricBatchSenderFactory.fromHttpImplementation(httpPosterCreator)
+                  .configureWith(insertApiKey)
+                  .build());
+
+      SpanBatchSender spanBatchSender =
+          SpanBatchSender.create(
+              SpanBatchSenderFactory.fromHttpImplementation(httpPosterCreator)
+                  .configureWith(insertApiKey)
+                  .build());
+
+      EventBatchSender eventBatchSender =
+          EventBatchSender.create(
+              EventBatchSenderFactory.fromHttpImplementation(httpPosterCreator)
+                  .configureWith(insertApiKey)
+                  .build());
+
+      long shutdownSeconds = 3L;
```

Comment:
Would be great if this could be promoted to a constant near the top of the file, something like `DEFAULT_SHUTDOWN_SECONDS`.